core/src/main/java/org/infinispan/util/functional/MarshallableFunctionalInterfaces.java

```diff
@@ -0,0 +1,341 @@
+import org.infinispan.commons.marshall.SerializeValueMatcher;
+import org.infinispan.commons.marshall.SerializeWith;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public final class MarshallableFunctionalInterfaces {
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, V> setValueReturnPrevOrNull() {
+      return SetValueReturnPrevOrNull.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, V> setValueIfAbsentReturnPrevOrNull() {
+      return SetValueIfAbsentReturnPrevOrNull.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, Boolean> setValueIfAbsentReturnBoolean() {
+      return SetValueIfAbsentReturnBoolean.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, V> setValueIfPresentReturnPrevOrNull() {
+      return SetValueIfPresentReturnPrevOrNull.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, Boolean> setValueIfPresentReturnBoolean() {
+      return SetValueIfPresentReturnBoolean.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, Boolean> setValueIfEqualsReturnBoolean(V oldValue) {
+      return new SetValueIfEqualsReturnBoolean<>(oldValue);
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> Function<ReadWriteEntryView<K, V>, V> removeReturnPrevOrNull() {
+      return RemoveReturnPrevOrNull.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> Function<ReadWriteEntryView<K, V>, Boolean> removeReturnBoolean() {
+      return RemoveReturnBoolean.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> BiFunction<V, ReadWriteEntryView<K, V>, Boolean> removeIfValueEqualsReturnBoolean() {
+      return RemoveIfValueEqualsReturnBoolean.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <V> BiConsumer<V, WriteEntryView<V>> setValueConsumer() {
+      return SetValue.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <V> Consumer<WriteEntryView<V>> removeConsumer() {
+      return Remove.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> Function<ReadWriteEntryView<K, V>, Optional<V>> findReadWrite() {
+      return FindReadWrite.INSTANCE;
+   }
+
+   @SuppressWarnings("unchecked")
+   public static <K, V> Function<ReadWriteEntryView<K, V>, V> getReadWrite() {
+      return GetReadWrite.INSTANCE;
+   }
+
+   @SerializeWith(value = SetValueReturnPrevOrNull.Externalizer0.class,
```

Comment:
So I am trying to understand SerializeWith a bit more.  It appears it always sends the externalizer with each serialized instance.  Why would we want this over an advanced externalizer?  Does this ship along the class to so we can add serialized lambdas at runtime?  Or was this so we have the valueMatcher?